dlls/winex11.drv/fs.c

```diff
@@ -535,6 +535,26 @@ BOOL fs_hack_is_integer(void)
     return is_int;
 }
 
+enum fs_hack_upscaler fs_hack_upscaler()
+{
+    static enum fs_hack_upscaler upscaler = X11_DRV_UPSCALER_NONE;
+
+    if (upscaler == X11_DRV_UPSCALER_NONE)
+    {
+        const char *e = getenv("WINE_FULLSCREEN_UPSCALER");
+        if (e && !strcmp(e, "NIS"))
+        {
+            upscaler = X11_DRV_UPSCALER_NIS;
+        }
+        else
```

Comment:
Maybe add a warning here for unrecognized values?

Reply:
Done.